### PR TITLE
Update type hints on part 10.

### PR DESCRIPTION
**Pull request by IRONICBo**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

mmdet/models/task_modules/coders/bucketing_bbox_coder.py
mmdet/models/task_modules/coders/delta_xywh_bbox_coder.py
mmdet/models/task_modules/coders/distance_point_bbox_coder.py
mmdet/models/task_modules/coders/legacy_delta_xywh_bbox_coder.py
mmdet/models/task_modules/coders/pseudo_bbox_coder.py
mmdet/models/task_modules/coders/tblr_bbox_coder.py
mmdet/models/task_modules/coders/yolo_bbox_coder.py

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/models/task_modules/coders/bucketing_bbox_coder.py [hunk 1]
```diff
@@ -71,7 +76,12 @@ def encode(self, bboxes, gt_bboxes):
                                      self.cls_ignore_neighbor)
         return encoded_bboxes
 
-    def decode(self, bboxes, pred_bboxes, max_shape=None):
+    def decode(
+            self,
+            bboxes: Union[Tensor, BaseBoxes],
+            pred_bboxes: Tensor,
+            max_shape: Optional[Tuple[int]] = None
+    ) -> Union[Tensor, BaseBoxes]:
```
Comment (hhaAndroid) [inline mmdet/models/task_modules/coders/bucketing_bbox_coder.py:84]:
```suggestion
    ) -> Tuple[Union[Tensor, BaseBoxes], Tensor]:
```

File: mmdet/models/task_modules/coders/delta_xywh_bbox_coder.py [hunk 2]
```diff
@@ -32,20 +34,21 @@ class DeltaXYWHBBoxCoder(BaseBBoxCoder):
...
     def __init__(self,
-                 target_means=(0., 0., 0., 0.),
-                 target_stds=(1., 1., 1., 1.),
-                 clip_border=True,
-                 add_ctr_clamp=False,
-                 ctr_clamp=32,
-                 **kwargs):
+                 target_means: Sequence[float] = (0., 0., 0., 0.),
+                 target_stds: Sequence[float] = (1., 1., 1., 1.),
+                 clip_border: Optional[bool] = True,
```
Comment (hhaAndroid) [inline mmdet/models/task_modules/coders/delta_xywh_bbox_coder.py:39]:
```suggestion
                 clip_border: bool = True,
```

Comment (IRONICBo) [reply mmdet/models/task_modules/coders/delta_xywh_bbox_coder.py]: Thanks :)

File: mmdet/models/task_modules/coders/bucketing_bbox_coder.py (hunk 1 above)
Comment (IRONICBo) [reply mmdet/models/task_modules/coders/bucketing_bbox_coder.py]: Thanks :)

File: mmdet/models/task_modules/coders/legacy_delta_xywh_bbox_coder.py [hunk 3]
```diff
@@ -139,12 +147,12 @@ def legacy_bbox2delta(proposals,
...
-                      deltas,
-                      means=(0., 0., 0., 0.),
-                      stds=(1., 1., 1., 1.),
-                      max_shape=None,
-                      wh_ratio_clip=16 / 1000):
+def legacy_delta2bbox(rois: Tensor,
+                      deltas: Tensor,
+                      means: Sequence[float] = (0., 0., 0., 0.),
+                      stds: Sequence[float] = (1., 1., 1., 1.),
+                      max_shape: Tuple[int, int] = None,
```
Comment (hhaAndroid) [inline mmdet/models/task_modules/coders/legacy_delta_xywh_bbox_coder.py:154]: max_shape: Optional[Union[Sequence[int], Tensor,
                                  Sequence[Sequence[int]]]] = None,

Comment (IRONICBo) [reply mmdet/models/task_modules/coders/legacy_delta_xywh_bbox_coder.py:154]: Thanks :)

File: mmdet/models/task_modules/coders/delta_xywh_bbox_coder.py [hunk 4]
```diff
@@ -168,15 +179,15 @@ def bbox2delta(proposals, gt, means=(0., 0., 0., 0.), stds=(1., 1., 1., 1.)):
...
-               max_shape=None,
-               wh_ratio_clip=16 / 1000,
-               clip_border=True,
-               add_ctr_clamp=False,
-               ctr_clamp=32):
+def delta2bbox(rois: Tensor,
+               deltas: Tensor,
+               means: Sequence[float] = (0., 0., 0., 0.),
+               stds: Sequence[float] = (1., 1., 1., 1.),
+               max_shape: Tuple[int, int] = None,
```
[the remark this answers is not in the document]
Comment (IRONICBo) [reply mmdet/models/task_modules/coders/delta_xywh_bbox_coder.py:186]: Thanks :)

File: mmdet/models/task_modules/coders/bucketing_bbox_coder.py [hunk 5]
```diff
@@ -268,13 +280,13 @@ def bbox2bucket(proposals,
...
-                scale_factor=1.0,
-                max_shape=None,
-                clip_border=True):
+def bucket2bbox(proposals: Tensor,
+                cls_preds: Tensor,
+                offset_preds: Tensor,
+                num_buckets: int,
+                scale_factor: float = 1.0,
+                max_shape: Tuple[int, int] = None,
+                clip_border: Optional[bool] = True) -> Tuple[Tensor]:
```
[the remark this answers is not in the document]
Comment (IRONICBo) [reply mmdet/models/task_modules/coders/bucketing_bbox_coder.py:289]: Thanks :)

File: mmdet/models/task_modules/coders/bucketing_bbox_coder.py [hunk 6]
```diff
@@ -32,13 +36,13 @@ class BucketingBBoxCoder(BaseBBoxCoder):
...
-                 offset_upperbound=1.0,
-                 cls_ignore_neighbor=True,
-                 clip_border=True,
-                 **kwargs):
+                 num_buckets: int,
+                 scale_factor: int,
+                 offset_topk: int = 2,
+                 offset_upperbound: float = 1.0,
+                 cls_ignore_neighbor: bool = True,
+                 clip_border: Optional[bool] = True,
```
[the remark this answers is not in the document]
Comment (IRONICBo) [reply mmdet/models/task_modules/coders/bucketing_bbox_coder.py:44]: Oh，Sorry, I have fixed it.

**Review by hhaAndroid — APPROVED** [review]

**Comment by OpenMMLab-Assistant005** [comment]
Hi @IRONICBo ！First of all, we want to express our gratitude for your significant PR in the MMDet project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**